src/Trace/TraceContext.php

```diff
@@ -0,0 +1,177 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Trace;
+
+/**
+ * TraceContext encapsulates your current context within your request's trace. It includes
+ * 3 fields: the `traceId`, the current `spanId`, and an `enabled` flag which indicates whether
+ * or not the request is being traced.
+ *
+ * Example:
+ *
+ * ```
+ * use Google\Cloud\Trace\RequestTracer;
+ *
+ * $context = RequestTracer::context();
+ * echo $context; // output the header format for using the current context in a remote call
+ * ```
+ */
+class TraceContext
+{
+    use IdGeneratorTrait;
+
+    const HTTP_HEADER = 'HTTP_X_CLOUD_TRACE_CONTEXT';
+    const CONTEXT_HEADER_FORMAT = '/([0-9a-f]{32})(?:\/(\d+))?(?:;o=(\d+))?/';
+
+    /**
+     * @var string The current traceId.
+     */
+    private $traceId;
+
+    /**
+     * @var string The current spanId. This is the deepest nested span currently open.
+     */
+    private $spanId;
+
+    /**
+     * @var bool Whether or not tracing is enabled for this request.
+     */
+    private $enabled;
+
+    /**
+     * Parses a headers array (normally the $_SERVER variable) and builds a TraceContext objects
+     *
+     * @param  array $headers The headers array (normally the $_SERVER variable)
+     * @return TraceContext
+     */
+    public static function fromHeaders($headers)
```

Comment:
Let's keep the parsing separate for now. I went down the rabbit hole of trying to share data between Trace/Logging/Error Reporting within Core, but it's really ugly because Trace/Logging/Error Reporting module may not even be loaded.

When the stackdriver API's stabilize, we can revisit sharing some of the environment detection.